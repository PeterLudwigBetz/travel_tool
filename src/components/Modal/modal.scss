.modal {
  position: absolute;
  top: 0;
  transform: translate(-50%, -50%);
  justify-content: center;
  align-items: center;
  background-color: #FFFFFF;
  z-index: 200;
  min-height: 180px;
  width: 100vw;
  max-width: 600px;
  min-width: 320px;
  box-sizing: border-box;
  font-family: "DIN Pro";
  font-weight: 800;
  color: #4F4F4F;
  outline: none;

  .modal-title-bar {
    position: relative;
    top: 0;
    display: flex;
    justify-content: space-between;
    align-items: center;
    width: 100%;
    height: 72px;
    box-sizing: border-box;
    padding: 0 30px;
    font-size: 21px;
    background-color: #F8F8F8;

    .modal-close {
      display: flex;
      justify-content: center;
      align-items: center;
      cursor: pointer;
      height: 17px;
      width: 17px;
      outline: none;
      border-style: none;
      background: none;
    }
  }

  .modal-content  {
    width: 100%;
    padding: 0 30px;
    box-sizing: border-box;
  }
}

@media (min-width: 640px) {
  .modal {
    top: 50px;
    margin-bottom: 50px;
  }
}
