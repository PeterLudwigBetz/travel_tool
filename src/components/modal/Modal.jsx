import React, { PureComponent, Fragment } from 'react';
import { PropTypes } from 'prop-types';
import Overlay from './overlay/Overlay';
import closeButton from '../../images/icons/close.svg';
import './_modal.scss';

class Modal extends PureComponent {
  static propTypes = {
    visibility: PropTypes.oneOf(['visible', 'invisible']).isRequired,
    toggleModal: PropTypes.func.isRequired,
    symbol: PropTypes.string,
    title: PropTypes.string,
    description: PropTypes.string,
    modalBar: PropTypes.string,
    children: PropTypes.oneOfType([
      PropTypes.object,
      PropTypes.arrayOf(PropTypes.object)
    ]).isRequired
    
  };

  renderModalHeader = () => {
    const { title, toggleModal, symbol, description, modalBar} = this.props;
    return (
      <div className="modal-title-bar">
        <div className="modal-title-text">
          <div>
            {symbol}
          </div>
          <div>
            {title}
          </div>
          <div className="modal-title-id">
            {description}
          </div>
          <div className="lable-text modal-bar">
            {modalBar}
          </div>
        </div>
        <button type="button" onClick={toggleModal} className="modal-close">
          <img alt="close" src={closeButton} />
        </button>
      </div>
    );
  };

  render() {
    const { children, visibility, toggleModal, title } = this.props;
    return (
      <Fragment>
        <Overlay click={toggleModal} className={visibility}>
          <div
            className={`modal ${visibility}`}
            onClick={e => {
              e.stopPropagation();
            }}
            onKeyPress={() => {}}
            tabIndex="0"
            role="button"
          >
            {this.renderModalHeader()}
            <div className="modal-content">
              {children}
            </div>
          </div>
        </Overlay>
      </Fragment>
    );
  }
}

Modal.defaultProps = {
  symbol: '',
  title: '',
  description: '',
  modalBar: ''
};

export default Modal;
