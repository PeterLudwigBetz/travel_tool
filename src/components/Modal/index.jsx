import React, { PureComponent, Fragment } from 'react';
import { PropTypes } from 'prop-types';
import Overlay from './overlay';
import closeButton from '../../images/icons/close.svg';
import './modal.scss';

class Modal extends PureComponent {
  static propTypes = {
    visibility: PropTypes.oneOf(['visible', 'invisible']).isRequired,
    title: PropTypes.string.isRequired,
    toggleModal: PropTypes.func.isRequired,
    children: PropTypes.oneOfType([
      PropTypes.object,
      PropTypes.arrayOf(PropTypes.object)
    ]).isRequired
  }

  renderModalHeader = () => {
    const {title, toggleModal} = this.props;
    return (
      <div className="modal-title-bar">
        <div className="modal-title-text">
          {title}
        </div>
        <button
          type="button"
          onClick={toggleModal}
          className="modal-close"
        >
          <img alt="close" src={closeButton} />
        </button>
      </div>
    );
  }

  render() {
    const {children, visibility, toggleModal, title} = this.props;

    return (
      <Fragment>
        <Overlay click={toggleModal} className={visibility}>
          <div
            className={`modal ${visibility}`}
            onClick={(e)=>{e.stopPropagation();}}
            onKeyPress={()=>{}}
            tabIndex="0"
            role="button"
          >
            {this.renderModalHeader()}
            <div className="modal-content">
              { children }
            </div>
          </div>
        </Overlay>
      </Fragment>
    );
  }
}

export default Modal;
