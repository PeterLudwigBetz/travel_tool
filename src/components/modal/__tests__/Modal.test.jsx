import React from 'react';
import Modal from '../Modal';

describe('<Modal />', () => {
  let props, click;

  beforeEach(() => {
    click = jest.fn();
    props = {
      toggleModal: click,
      title: 'test modal',
      visibility: 'visible'
    };
  });

  it('renders correctly', () => {
    const wrapper = shallow(
      <Modal {...props}>
        <div>
          Test content
        </div>
      </Modal>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('closes when close button is clicked', () => {
    const handleClose = jest.fn();
    const wrapper = mount(
      <Modal {...props} toggleModal={handleClose}>
        <div>
          Test content
        </div>
      </Modal>
    );
    wrapper
      .find('button.modal-close')
      .simulate('click');
    expect(handleClose).toHaveBeenCalledTimes(1);
  });
});
