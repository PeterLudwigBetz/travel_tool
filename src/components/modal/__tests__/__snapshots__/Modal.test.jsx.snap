// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> renders correctly 1`] = `
<Fragment>
  <Overlay
    className="visible"
    click={null}
  >
    <div
      className="modal visible"
      id=""
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      style={
        Object {
          "maxWidth": "",
        }
      }
      tabIndex="0"
    >
      <div
        className="modal-title-bar"
      >
        <div
          className="modal-title-text"
        >
          <div>
            test modal
          </div>
          <div
            className="lable-text modal-bar"
          >
            <div />
          </div>
        </div>
        <Link
          replace={false}
          to="/"
        >
          <button
            className="modal-close"
            onClick={[Function]}
            type="button"
          >
            <img
              alt="close"
              src="close.svg"
            />
          </button>
        </Link>
      </div>
      <div
        className="modal-content"
        id=""
      >
        <div>
          Test content
        </div>
      </div>
    </div>
  </Overlay>
</Fragment>
`;
